docs/cache_proposal.md

```diff
@@ -0,0 +1,223 @@
+# CZ-Benchmarks Storage and Caching
+
+# Introduction
+
+The cz-benchmarks package needs to be able to make use of large filesize models and datasets stored in a variety of different systems. Currently, the CLI makes use of caching the results of inference to save computation, and it would be nice to continue supporting that. In the long term, cz-benchmarks might need to be able to cache models fine-tuned for particular benchmarking tasks, or checkpoints created during fine-tuning.
```

Comment:
a couple more to add to this list of data the system would handle:
* task results [1]
* benchmarking datasets transformed into data model inputs (keyed to a specific model)

[1] I think of task results as "stored" data, rather than "cached" data, but "Storage and Caching" system name covers both cases.

Reply:
+1 to the idea of thinking about both of storage and caching as the same thing